tests/mgbench/benchmark.py

```diff
@@ -58,71 +205,281 @@ def filter_benchmarks(generators, patterns):
+                # Get the apropropriate functions with same probabilty
+                if t == "query":
+                    full_workload.append(base_query())
+                else:
+                    funcname = random.choices(queries_by_type[t], k=1)[0]
+                    aditional_query = getattr(dataset, funcname)
+                    full_workload.append(aditional_query())
+
+            vendor.start_benchmark(
+                dataset.NAME + dataset.get_variant() + "_" + "mixed" + "_" + query + "_" + config_distribution
+            )
+            if args.warmup_run:
+                warmup(client)
+            ret = client.execute(
+                queries=full_workload,
+                num_workers=args.num_workers_for_benchmark,
+            )[0]
+            usage_workload = vendor.stop(
+                dataset.NAME + dataset.get_variant() + "_" + "mixed" + "_" + query + "_" + config_distribution
+            )
+
+            ret["database"] = usage_workload
+
+            results_key = [
+                dataset.NAME,
+                dataset.get_variant(),
+                group,
+                query + "_" + config_distribution,
+                WITHOUT_FINE_GRAINED_AUTHORIZATION,
+            ]
+            results.set_value(*results_key, value=ret)
+
+    else:
+        # Executing mixed workload from groups of queries
+        full_workload = []
+        options = ["write", "read", "update", "analytical"]
+        function_type = random.choices(population=options, weights=percentage_distribution, k=num_of_queries)
+
+        for t in function_type:
+            # Get the apropropriate functions with same probabilty
+            funcname = random.choices(queries_by_type[t], k=1)[0]
+            aditional_query = getattr(dataset, funcname)
+            full_workload.append(aditional_query())
+
+        vendor.start_benchmark(dataset.NAME + dataset.get_variant() + "_" + workload.name + "_" + config_distribution)
+        if args.warmup_run:
+            warmup(client)
+        ret = client.execute(
+            queries=full_workload,
+            num_workers=args.num_workers_for_benchmark,
+        )[0]
+        usage_workload = vendor.stop(
+            dataset.NAME + dataset.get_variant() + "_" + workload.name + "_" + config_distribution
+        )
+        mixed_workload = {
+            "count": ret["count"],
+            "duration": ret["duration"],
+            "retries": ret["retries"],
+            "throughput": ret["throughput"],
+            "num_workers": ret["num_workers"],
+            "database": usage_workload,
+        }
+        results_key = [
+            dataset.NAME,
+            dataset.get_variant(),
+            group,
+            config_distribution,
+            WITHOUT_FINE_GRAINED_AUTHORIZATION,
+        ]
+        results.set_value(*results_key, value=mixed_workload)
+
+        print(mixed_workload)
+
+
+def get_query_cache_count(vendor, client, func, config_key):
```

Comment:
Introduce "time-constrained execution" + this "query number constrained execution"